### PR TITLE
redeclare examples as integration tests

### PR DESCRIPTION
**Pull request by BenjaminW3**
As has been noted in #77 the current examples are not really easy examples but represent more or less integration tests. Therefore they have been moved to `test/integ`.

### CODE REVIEW

**Comment by BenjaminW3** [comment]
I have additionally added a vector addition single file example without additional includes using a single accelerator and without time measurement.

**Comment by BenjaminW3** [comment]
When this is merged, I will add a first unit test for #79/#85 using boost.test.

**Comment by psychocoderHPC** [comment]
Could you please split this pull request one for move examples to integration tests and one for the first easy example. It is easier to review.

File: example/vectorAdd/src/main.cpp [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+        auto const gridThreadIdx(alpaka::idx::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u]);
+        auto const threadElemExtent(alpaka::workdiv::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[0u]);
+        auto const threadFirstElemIdx(gridThreadIdx * threadElemExtent);
+
+        if(threadFirstElemIdx < numElements)
+        {
+            // Calculate the number of elements to compute in this thread.
+            // The result is uniform for all but the last thread.
+            auto const elems(threadElemExtent + alpaka::math::min(acc, static_cast<unsigned long long>(0u), static_cast<unsigned long long>(numElements-(threadFirstElemIdx+threadElemExtent))));
```
Comment (psychocoderHPC) [inline example/vectorAdd/src/main.cpp]: independent of the point that we should move this in a separate pull request:

Might be we should create a alpaka trait/function for this line.

**Comment by BenjaminW3** [comment]
The request has been split.

File: travis/compileExec.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    -DALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLE="${ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLE}" -DALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLE="${ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLE}" -DALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE="${ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE}" -DALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE="${ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE}" -DALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE="${ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE}" -DALPAKA_ACC_GPU_CUDA_ENABLE="${ALPAKA_ACC_GPU_CUDA_ENABLE}" \
+    -DALPAKA_DEBUG="${ALPAKA_DEBUG}" -DALPAKA_INTEGRATION_TEST=ON -DALPAKA_CUDA_VERSION="${ALPAKA_CUDA_VERSION}" \
+    "../../"
+make VERBOSE=1
+if [ "${ALPAKA_ACC_GPU_CUDA_ENABLE}" == "OFF" ]
+then
+    eval "${2}"
+fi
+
+cd "${oldPath}"
```
Comment (psychocoderHPC) [inline travis/compileExec.sh]: missing end of line

**Comment by psychocoderHPC** [comment]
after the missing EOF is added I can merge this pull request

File: travis/compileExec.sh (hunk 2 above)
Comment (BenjaminW3) [inline travis/compileExec.sh]: Fixed.
